### ISSUE TITLE
ms-swift 4.0 lr_scheduler_type报错

### ISSUE THREAD

**Issue by hpsun1109** [modelscope/ms-swift#8172]
### Checklist / 检查清单

- [x] I have searched existing issues, and this is a new bug report. / 我已经搜索过现有的 issues，确认这是一个新的 bug report。

### Bug Description / Bug 描述

   --lr_scheduler_type cosine_with_min_lr \
    --lr_scheduler_kwargs '{"min_lr": 1e-6}' \
加上上述参数会报错

<img width="969" height="539" alt="Image" src="https://github.com/user-attachments/assets/735094ea-3ba4-4634-8335-0fcd9f16ca1b" />

### How to Reproduce / 如何复现

--lr_scheduler_type cosine_with_min_lr
--lr_scheduler_kwargs '{"min_lr": 1e-6}'
加上上述参数会报错

**Comment by Jintao-Huang** [comment]
transformers和accelerate版本是什么

swift没有改这块

**Comment by hpsun1109** [comment]
> transformers和accelerate版本是什么
> 
> swift没有改这块

transformers是5.2.0
accelerate是1.12.0

**Comment by qy0110** [comment]
同上🙋，遇到了相同的问题，请问要怎么解决啊？

**Comment by HJXA** [comment]
相同的问题，在使用deepspeed时会报错，fsdp不会

**Comment by HJXA** [comment]
> 同上🙋，遇到了相同的问题，请问要怎么解决啊？



> ### Checklist / 检查清单
> * [x]  I have searched existing issues, and this is a new bug report. / 我已经搜索过现有的 issues，确认这是一个新的 bug report。
> 
> ### Bug Description / Bug 描述
> --lr_scheduler_type cosine_with_min_lr --lr_scheduler_kwargs '{"min_lr": 1e-6}' 加上上述参数会报错
> 
> <img alt="Image" width="969" height="539" src="https://private-user-images.githubusercontent.com/6996472/557340731-735094ea-3ba4-4634-8335-0fcd9f16ca1b.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzY0ODk2NzYsIm5iZiI6MTc3NjQ4OTM3NiwicGF0aCI6Ii82OTk2NDcyLzU1NzM0MDczMS03MzUwOTRlYS0zYmE0LTQ2MzQtODMzNS0wZmNkOWYxNmNhMWIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI2MDQxOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNjA0MThUMDUxNjE2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OWJjZmM2NDU5Zjg5ZDRkOWIxMWVjNzdlZDAzZjg4ZmVjNGM1MzhjNTY5MzdkZTIzMTAwYzE2NDZhYjE3ZDExNyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmcmVzcG9uc2UtY29udGVudC10eXBlPWltYWdlJTJGcG5nIn0.Rjf9eIbFh00PByJSI5nhLuf2BSDBMEMtXb3AFbzfP7U">
> ### How to Reproduce / 如何复现
> --lr_scheduler_type cosine_with_min_lr --lr_scheduler_kwargs '{"min_lr": 1e-6}' 加上上述参数会报错
> 
> ### Additional Information / 补充信息
> _No response_



> transformers和accelerate版本是什么
> 
> swift没有改这块

我的解决方案如下：
我将'{"min_lr": xxx}'改为'{"min_lr_rate": xxx}'就不会报错

原因猜测：
```python
# optimization.py
if min_lr is not None and min_lr_rate is not None:
        raise ValueError("Only one of min_lr or min_lr_rate should be set")
    elif min_lr is not None:
        min_lr_rate = min_lr / optimizer.defaults["lr"]
    elif min_lr_rate is None:
        raise ValueError("One of min_lr or min_lr_rate should be set through the `lr_scheduler_kwargs`")
```
这应该是因为deepspeed下，其没有设定原始的lr

**Comment by daixiangzi** [comment]
上面设置是不对的，min_lr和min_lr_rate意义完全不一样

<img width="2236" height="1492" alt="Image" src="https://github.com/user-attachments/assets/d4f81ad6-7705-4d8b-b4ed-5cdda47674ea" />

**Comment by daixiangzi** [comment]
要避开zero3的bug,可能需要min_lr_rate = min_lr / optimizer.defaults["lr"]（这样自己转换一下）

**Comment by HJXA** [comment]
> 要避开zero3的bug,可能需要min_lr_rate = min_lr / optimizer.defaults["lr"]（这样自己转换一下）

是的，我之前的回答有问题

**Comment by HJXA** [comment]
> 要避开zero3的bug,可能需要min_lr_rate = min_lr / optimizer.defaults["lr"]（这样自己转换一下）

<img width="2752" height="1124" alt="Image" src="https://github.com/user-attachments/assets/5dfc8c69-bde0-4770-8f99-c1662ee9af9f" />
但是我的实际的训练的学习率结果如上，我还是遵照的之前的设置，其好像满足的就是 min_lr = lr * min_lr_rate,而不是像代码注释的那样是inital learning rate

**Comment by daixiangzi** [comment]
> > 要避开zero3的bug,可能需要min_lr_rate = min_lr / optimizer.defaults["lr"]（这样自己转换一下）
> 
> <img alt="Image" width="2000" height="817" src="https://private-user-images.githubusercontent.com/182237060/593449339-5dfc8c69-bde0-4770-8f99-c1662ee9af9f.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzkwNzIyOTcsIm5iZiI6MTc3OTA3MTk5NywicGF0aCI6Ii8xODIyMzcwNjAvNTkzNDQ5MzM5LTVkZmM4YzY5LWJkZTAtNDc3MC04Zjk5LWMxNjYyZWU5YWY5Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNTE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDUxOFQwMjM5NTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yOTU3MWE1NDFiMmM4ZjkxOTAwNDNiOTBhZTVjODc1NWJlZDk0YTEyOWRhZTc5ZTlkNDMzNGZjM2JmOWRjN2FmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZyZXNwb25zZS1jb250ZW50LXR5cGU9aW1hZ2UlMkZwbmcifQ.PNxYQRWOfpZgUCa_44zxPZ6MUdVTqW24czZk_y12_G8"> 但是我的实际的训练的学习率结果如上，我还是遵照的之前的设置，其好像满足的就是 min_lr = lr * min_lr_rate,而不是像代码注释的那样是inital learning rate

我刚才看，最新的版本好像修复了这个问题，你可以看看
